events/tests/test_organization_api.py

```diff
@@ -1,247 +1,399 @@
-    url = reverse("organization-list")
+def test_user_is_automatically_added_to_admins_users(api_client, data_source, user):
     api_client.force_authenticate(user)
 
+    origin_id = "test_organization2"
     payload = {
-        "data_source": organization.data_source.id,
-        "id": organization.id,
-        "name": "new name",
-        "origin_id": "test_organization",
+        "data_source": data_source.id,
+        "origin_id": origin_id,
+        "id": f"{data_source.id}:{origin_id}",
+        "name": "test org",
     }
 
-    response = api_client.put(f"{url}{organization.id}/", payload, format="json")
-    assert response.data["name"] == payload["name"]
+    response = assert_create_organization(api_client, payload)
+    assert response.data["admin_users"][0]["username"] == user.username
 
 
 @pytest.mark.django_db
-def test_put_user_has_no_rights(user, user2, organization, api_client):
-    url = reverse("organization-list")
-    api_client.force_authenticate(user2)
+def test_admin_user_add_users_to_new_organization(api_client, data_source, user, user2):
+    api_client.force_authenticate(user)
 
+    origin_id = "test_organization2"
     payload = {
-        "data_source": organization.data_source.id,
-        "id": organization.id,
-        "name": "new name",
-        "origin_id": "test_organization",
+        "data_source": data_source.id,
+        "origin_id": origin_id,
+        "id": f"{data_source.id}:{origin_id}",
+        "name": "test org",
+        "admin_users": [user.username, user2.username],
+        "regular_users": [user2.username],
     }
 
-    response = api_client.put(f"{url}{organization.id}/", payload, format="json")
-    assert response.status_code == 403
+    response = assert_create_organization(api_client, payload)
+    assert set(payload["admin_users"]) == set(
+        [i["username"] for i in response.data["admin_users"]]
+    )
+    assert set(payload["regular_users"]) == set(
+        [i["username"] for i in response.data["regular_users"]]
+    )
 
 
 @pytest.mark.django_db
-def test_delete_user_has_rights(user, organization, api_client):
-    url = reverse("organization-list")
+def test_admin_user_can_update_organization(api_client, organization, user):
     api_client.force_authenticate(user)
 
-    response = api_client.delete(f"{url}{organization.id}/")
-    assert response.status_code == 204
-    response = api_client.get(f"{url}{organization.id}/", format="json")
-    assert response.status_code == 404
+    payload = {
+        "id": organization.id,
+        "name": "new name",
+    }
+
+    response = assert_update_organization(api_client, organization.id, payload)
+    assert response.data["name"] == payload["name"]
 
 
 @pytest.mark.django_db
-def test_delete_user_has_no_rights(user, user2, organization, api_client):
-    url = reverse("organization-list")
-    api_client.force_authenticate(user2)
+def test_anonymous_user_can_update_organization(api_client, organization):
```

Comment:
Probably should be `test_anonymous_user_cannot_update_organization`

Reply:
Good catch. Fixed test name